cmake/leatherman.cmake

```diff
@@ -131,6 +131,90 @@ macro(enable_cpplint)
     endif()
 endmacro()
 
+# Usage: gettext_templates(dir)
+#
+# Create templates for gettext in `dir` from the source files found in additional specified dirs
+# Creates a custom target `translation`.
+macro(gettext_templates dir)
+    find_program(XGETTEXT_EXE xgettext)
+    find_program(MSGINIT_EXE msginit)
+    find_program(MSGMERGE_EXE msgmerge)
+
+    if (XGETTEXT_EXE AND MSGINIT_EXE AND MSGMERGE_EXE)
+        set(TRANSLATION_DIR "${dir}")
+        file(MAKE_DIRECTORY ${TRANSLATION_DIR})
+
+        file(GLOB_RECURSE ALL_PROJECT_SOURCES ${ARGN})
+        set(lang_template ${TRANSLATION_DIR}/${PROJECT_NAME}.pot)
+        execute_process(COMMAND ${XGETTEXT_EXE}
+            --c++ --boost -F
+            --copyright-holder "Puppet Labs <docs@puppetlabs.com>"
+            --package-name=${PROJECT_NAME} --package-version=${PROJECT_VERSION}
+            --msgid-bugs-address "docs@puppetlabs.com"
+            -d ${PROJECT_NAME} -o ${lang_template}
+            --keyword=LOG_DEBUG:1,"debug"
+            --keyword=LOG_INFO:1,"info"
+            --keyword=LOG_WARNING:1,"warning"
+            --keyword=LOG_ERROR:1,"error"
+            --keyword=LOG_FATAL:1,"fatal"
+            --keyword=_
+            --keyword=translate
+            --keyword=format
+            ${ALL_PROJECT_SOURCES})
+
+        file(READ ${lang_template} FILE_CONTENT)
+        string(REPLACE "charset=CHARSET" "charset=UTF-8"
+            FILE_CONTENT "${FILE_CONTENT}")
+        string(REGEX REPLACE "POT-Creation-Date: [^\\]*" "POT-Creation-Date: "
+            FILE_CONTENT "${FILE_CONTENT}")
+        string(REPLACE "${CMAKE_CURRENT_SOURCE_DIR}/" "" FILE_CONTENT "${FILE_CONTENT}")
+        file(WRITE ${lang_template} "${FILE_CONTENT}")
+
+        file(GLOB existing_files ${TRANSLATION_DIR}/*.po)
```

Comment:
Don't manage .po files? Translation workflow assumes they'll monitor project .pot files and generate updated .po files.